src/main/java/pl/nogacz/snake/board/Board.java

```diff
@@ -116,6 +172,98 @@ private void moveSnakeHead(Coordinates coordinates) {
         }
     }
 
+    private boolean spawnRottenApple(int i){ // control for randomly chosen spawning rotten apple time

```

Comment:
Bu kodlar Board classını çok şişirmiş. Yaptığın değişiklikleri yeni bir class a taşı ve o classın metotlarını buradan çağır.

PRında çok fazla tekrar eden kod var. Aynı kod bloklarını 3er kez aynen tekrar ediyorsun. Kodun sade klaması benim için önemli. Tekrar eden kodlara bak ve bunları çeşitli metotlara taşıyarak reuse yap.

Mesela hem aşşağıda aynı if bloğu 3 kez geçiyor. Aradaki fark yalnızca 0, 1, 2 indexleri ve nextInt parametreleri. Bir metodu bu değerlerle çağırabilirsin üç defa bir döngüde.

Genel olarak bu dediğime dikkat et. Bir de gerekirse 10 çeşit yeni özel itemı kolaylıkla ekleyebileceğimiz bir şekilde kod esnek tasarlanmış olmalı.

Bu sebeple senden istediğim bu tür objelerle ilgili bir class hiyerarşisi yaratman. Bu 3 çeşit itemi de o classın 3 farklı sublassı olarak tanımlaman. Daha sonra ortak olan kodları hiyerarşide top classa koyarsın ve oldukça bu tür tekrarlarda azalma olacaktır.